client/components/FormattedText/FormattedText.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import styles from './FormattedText.module.scss';
+
+type FormattedTextProps = {
+  as?: React.ComponentType | keyof JSX.IntrinsicElements;
+  children?: React.ReactNode;
+} & React.AllHTMLAttributes<HTMLOrSVGElement>;
+
+const FormattedText: React.FC<FormattedTextProps> = ({
+  as: Wrapper = 'div',
+  children,
+  ...props
+}) => {
+  return (
+    <Wrapper {...props} className={styles.formatted}>
```

Comment:
I get a TS error here:

![image](https://user-images.githubusercontent.com/11327761/125408639-cb11c780-e3c3-11eb-98f9-166a96bc75a0.png)

I think it's because of this: `React.AllHTMLAttributes<HTMLOrSVGElement>`. Did you try having a look at https://stackoverflow.com/questions/55969769/typing-a-dynamic-tag-in-react-with-typescript?

Reply:
Yes i also got that error but to avoid it you have to use `string` type in `as` prop which prevents the emmet suggestions of html elements while using the component.. so i couldn't decide what to do hence gave preference to emmit suggestions

Reply:
I see. I think we can simplify the `as` prop: `as?: keyof JSX.IntrinsicElements;`. We don't really care about other components now, only standard HTML elements.

Then in code I think this is what we want: https://stackoverflow.com/a/65865806/10437751

Seems to be working:

```
<MyComponent<'form'> tag="span" noValidate> Ceva </MyComponent>
<MyComponent<'form'> tag="button" type="button"> Ceva </MyComponent>
```

The `type="button"` is throwing an error because `form` elements don't have `type` attribute :D

Reply:
Although having to add `<Type>` is a little bit weird, seems to be doing the job.